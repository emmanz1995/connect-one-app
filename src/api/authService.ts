import axios from 'axios';

async function onSignIn(email: string, password: string) {
  let response
  try {
    const payload = {
      email,
      password
    }
    response = await axios({
      url: `${import.meta.env.VITE_API_URL}/api/auth`,
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      data: payload,
    })
    console.log('...email2', email)
    console.log('...password2', password)
  } catch(err) {
    console.error(err)
    throw err
  }

  return response.data;
}

export const authService = {
  onSignIn
}