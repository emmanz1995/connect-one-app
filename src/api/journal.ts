import axios from 'axios';

const getJournal = async () => {}

const getMyJournals = async () => {}

const getSingleJournal = async (id: string | undefined) => {}

const onCreateJournal = async(formData: { title: string, description: string }) => {}

const onDeleteJournal = async (id: string) => {
  try {
    const { data } = await axios({
      url: `${import.meta.env.VITE_API_URL}/api/journal/${id}`,
      method: 'DELETE'
    })
    return data;
  } catch (err) {
    console.log(err);
    throw err;
  }
}

export {
  getJournal,
  getSingleJournal,
  onCreateJournal,
  onDeleteJournal
}