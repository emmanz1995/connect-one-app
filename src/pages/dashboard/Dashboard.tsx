import { useEffect, useState } from 'react';
import Navbar from '../../components/navbar/Navbar';
import Sidebar from '../../components/sidebar/Sidebar';
// import './dashboard.scss';
import * as api from '../../api/journal';
import { map } from 'lodash';
import Card from '../../components/Card/Card';
import CreateJournal from '../../components/createJournal/CreateJournal';

interface IJournal {
  title: string;
  description: string;
  completedAt: boolean;
  createdAt: string;
  updatedAt: string;
  id: string;
}

function Dashboard() {
  const [ journals, setJournals ] = useState<IJournal[]>([]);
  // const [ message, setMessage ] = useState<string>('');

  useEffect(() => {
    let isMounted = true

    async function fetchJournals() {
      try {
        const journalEntries: IJournal[] = await api.getJournal();
        if(isMounted) setJournals(journalEntries);
      } catch(err: any) {
        console.log(err)
      }
    }
    fetchJournals();

    return () => {
      isMounted = false;
    }
  }, []);

  const handleAddJournal = async ({ title, description }) => {
    try {
      const newJournal = await api.onCreateJournal({
        title,
        description
      })
      setJournals([ ...journals, newJournal ]);

      console.log('New Journal Created:', newJournal);
    } catch(err: unknown) {
      console.log(err)
    }
  }

  const mappedJournals = map(journals, (journal: IJournal) => (
    <div className="main__journal" key={journal?.id}>
      <Card journal={journal} />
    </div>
  ));
  return (
    <>
      <Navbar />
      <div className="main">
        <div className="main__menu">
          <Sidebar />
        </div>
        <div className="main__hub">
          <h4>Dashboard</h4>
          <CreateJournal
            onAddJournal={handleAddJournal}
          />
          <div className="main__journalSection">
            {mappedJournals}
          </div>
        </div>
      </div>
    </>
  )
}

export default Dashboard;
