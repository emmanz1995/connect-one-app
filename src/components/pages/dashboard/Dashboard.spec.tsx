import { create } from 'react-test-renderer';
import { render, act, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Dashboard from './Dashboard';
import * as api from '../../../api/journal';

// Mock the dependencies used in the Dashboard component
jest.mock('../../../api/journal');

const mockJournal = {
  title: 'Homer Simpson',
  description: 'Homer Simpson is a family man!',
  createdAt: new Date(),
  updatedAt: new Date(),
  id: '643db8c388f22f9d7395a0f5'
}

describe('Dashboard component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  let component
  let tree
  beforeEach(() => {
    component = create(<Dashboard />)
    tree = component.toJSON()
  })

  // Snapshot test for the Dashboard component
  it('renders correctly', () => {
    (api.getJournal as jest.MockedFunction<typeof api.getJournal>).mockResolvedValue([mockJournal])
    expect(tree).toMatchSnapshot();
  });

  it('should render component', async () => {
    (api.getJournal as jest.MockedFunction<typeof api.getJournal>).mockResolvedValue([mockJournal])

    await act(async () => {
      render(<Dashboard />)
    });
    expect(screen.getByText('Homer Simpson')).toBeInTheDocument
  });

  it('should add a new journal entry', async () => {
    (api.getJournal as jest.MockedFunction<typeof api.getJournal>).mockResolvedValue([]);
    (api.onCreateJournal as jest.MockedFunction<typeof api.onCreateJournal>).mockResolvedValue(mockJournal);

    render(<Dashboard />)

    await userEvent.type(screen.getByText('Title'), 'Homer Simpson')
    await userEvent.type(screen.getByText('Description'), 'Homer Simpson is a family man!')
    await userEvent.click(screen.getByText('Submit'))

    await waitFor(() => expect(screen.getByText('Homer Simpson')).toBeInTheDocument)
  });
});