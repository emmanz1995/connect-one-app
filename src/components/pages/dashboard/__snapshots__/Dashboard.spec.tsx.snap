// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders correctly 1`] = `
[
  <div
    className="navbar"
  >
    <div
      className="title"
    >
      <h3>
        Hello World!
      </h3>
    </div>
  </div>,
  <div
    className="main"
  >
    <div
      className="main__menu"
    >
      <div
        className="sidebar"
      >
        <div
          className="title"
        >
          <h4>
            Hello World!
          </h4>
        </div>
      </div>
    </div>
    <div
      className="main__hub"
    >
      <h4>
        Dashboard
      </h4>
      <form
        className="createJournal"
      >
        <div
          className="createJournal__header"
        >
          <h4>
            Create new journal
          </h4>
        </div>
        <div>
          <label
            htmlFor="title"
          >
            Title
          </label>
          <input
            className="createJournal__input"
            id="title"
            name="title"
            onChange={[Function]}
            placeholder="Your prayer journal title"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            htmlFor="description"
          >
            Description
          </label>
          <input
            className="createJournal__input"
            id="description"
            name="description"
            onChange={[Function]}
            placeholder="Your prayer journal description"
            type="text"
            value=""
          />
        </div>
        <button
          onClick={[Function]}
        >
          Submit
        </button>
      </form>
      <div
        className="main__journalSection"
      />
    </div>
  </div>,
]
`;
